### PR TITLE
MBS-8301: Fix the cleaning of relationships link phrases in the editor interface

### PR DESCRIPTION
**Pull request by y-van-z**
Remove spurious spaces left from relationships attributes, especially disturbing for tree indentation.

### CODE REVIEW

**Comment by uklauer** [comment]
Please give an explanation of what you are doing (in the commit message, in particular). I can take a wild guess that there are spurious spaces in the relationship description that get removed by `trim`, but I have no idea why that would be.

**Comment by y-van-z** [comment]
Sorry, I did it in JIRA instead, see [this comment](http://tickets.musicbrainz.org/browse/MBS-8301?focusedCommentId=40109&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#action_40109). I will fix git messages.

**Comment by y-van-z** [comment]
I changed the commit code for a more general fix to the `linkPhrase.clean` function.

**Comment by y-van-z** [comment]
This more general fix impacts the website user interface for non editors.
Using `replace` rather than `trim` for older browsers (IE 8).

**Comment by uklauer** [comment]
I think I understand what’s happening now. (I don’t think the commit message is as detailed as it needs to be. It should explain what the broken code is supposed to do, for context; i.e. removing attribute placeholders from link phrases.) However, as I understand it, there could still remain internal sequences of multiple spaces (`foo {additional|additionally} bared by` → `foo  bared by`), and that should also be fixed. 

Regarding `trim`, a) we don’t care about IE 8 for editing, only for viewing, b) we have lodash (`_`).

**Comment by y-van-z** [comment]
Now cleaning internal sequences of contiguous whitespaces too. Improved message. Thanks!

File: root/static/scripts/edit/utility/linkPhrase.js [hunk 1]
```diff
@@ -31,7 +31,7 @@ exports.clean = _.memoize(function (linkType, backward) {
         }
 
         return match;
-    });
+    }).trim().replace(/ {2,}/g, ' ');
```
Comment (mwiencek) [inline root/static/scripts/edit/utility/linkPhrase.js]: That's actually what https://github.com/y-van-z/musicbrainz-server/blob/a486351bec11911b1738d3c09fa251970a4d06a7/root/static/scripts/common/utility/clean.js does and it's already imported in this file. So you can just wrap the return like `return clean(phrase.replace(...));`.

**Comment by y-van-z** [comment]
Cleaner 7-characters patch proposed by [mwiencek’s comment](https://github.com/metabrainz/musicbrainz-server/pull/345/files/a486351bec11911b1738d3c09fa251970a4d06a7#r72913888). Thanks.

**Comment by mwiencek** [comment]
Thank you!

**Pull request merged**